### DISCUSSION TITLE
How to calculate the horizontal gradient in longitude/latitude coordinates?

### DISCUSSION THREAD

**Discussion by NMC-DAVE** [SciTools/iris#4777] [Q&A]
## 📰 Custom Issue
Hi,
       I want to calculate the horizontal gradient in longitude/latitude coordinates. So can you show me
an example to do this? Thanks very much!
                                                                 Dave

**Comment by trexfeathers** [comment]
Hi @NMC-DAVE, thanks for getting in touch.

Iris has no functionality for horizontal temperature gradients, but it should be achievable:

* I expect using [`numpy.gradient()`](https://numpy.org/doc/stable/reference/generated/numpy.gradient.html) on the `Cube.data` array will give you the raw numbers you need.
* You'll need to derive the coordinates (since each point will be half-way between the original points?).
  * You could use [`numpy.interp()`](https://numpy.org/doc/stable/reference/generated/numpy.interp.html) on the `DimCoord.points` arrays for this.
  * Create new `DimCoord`s using the resulting arrays.
* Create a new `Cube` using the array from the first step, and the new `DimCoord`s from the second step.

**Comment by pp-mo** [comment]
> @trexfeathers Iris has no functionality for horizontal temperature gradients

I'm not totally sure what you mean, though.
Would [iris.analysis.calculus.differentiate](https://scitools-iris.readthedocs.io/en/latest/generated/api/iris/analysis/calculus.html#iris.analysis.calculus.differentiate) be of any use in your case ?

If so, I think it does do some of the work to make new 'derived coordinates', as just mentioned.

**Reply by NMC-DAVE** [reply]
Very thanks @pp-mo ,
           [iris.analysis.calculus.differentiate](https://scitools-iris.readthedocs.io/en/latest/generated/api/iris/analysis/calculus.html#iris.analysis.calculus.differentiate) may do the job if the cube coordinates are an equal distance like 5km delta. But for longitude/latitude coordinates, I have to calculate the temperature gradient on spheric coordinates with an unequal delta. The Metpy has [metpy.calc.lat_lon_grid_deltas](https://unidata.github.io/MetPy/latest/api/generated/metpy.calc.lat_lon_grid_deltas.html) function to compute. So iris have the same function. Thanks again.
                   Dave
